fixes https://github.com/ssl-hep/X509_Secrets/issues/10

ivukotic opened the pull request:
original code was never actually refreshing proxy file, it was also trying to recreate secret that was already there and failing, so pod would restart every 6h.

Commit:
fixes https://github.com/ssl-hep/X509_Secrets/issues/10

Commit:
moved to osg 3.5. it already has python3.6

gordonwatts commented:
This looks like it had major changes to the setup, the OSG version, etc. Has it been tested running?

gordonwatts commented:
I also added @BenGalewsky to the list as I think he was one of the original authors for these files.

ivukotic commented:
Hi,

I have tested it, but one more person doing a test is always good to have.
I updated OSG version. This version already has python3.6 so I cleaned up the image.
Also I changed the code remaking the secret.

Cheers,
Ilija


From: Gordon Watts <notifications@github.com>
Sent: Wednesday, January 20, 2021 12:44
To: ssl-hep/X509_Secrets <X509_Secrets@noreply.github.com>
Cc: Ilija Vukotic <ivukotic@uchicago.edu>; Author <author@noreply.github.com>
Subject: Re: [ssl-hep/X509_Secrets] fixes https://github.com/ssl-hep/X509_Secrets/issues/10 (#11)


This looks like it had major changes to the setup, the OSG version, etc. Has it been tested running?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://urldefense.com/v3/__https:/github.com/ssl-hep/X509_Secrets/pull/11*issuecomment-763853507__;Iw!!BpyFHLRN4TMTrA!vgqIrVjWfQq5u1Q7M-vRtO4yBOYICc1jgNcwMI5Vl0K38-3rBsi9v6_iKFTgtDNzYZ0s$>, or unsubscribe<https://urldefense.com/v3/__https:/github.com/notifications/unsubscribe-auth/AANICGHXB56MAHXFS5NATDLS24P5NANCNFSM4WAGUFPQ__;!!BpyFHLRN4TMTrA!vgqIrVjWfQq5u1Q7M-vRtO4yBOYICc1jgNcwMI5Vl0K38-3rBsi9v6_iKFTgtDdhetNQ$>.

BenGalewsky commented:
I'll take a look at it this afternoon. Was waiting for some stability on River, but this looks like a good step forward for the whole stack.

BenGalewsky commented on Dockerfile:
```diff
@@ -47,5 +30,5 @@ RUN echo "Timestamp:" `date --utc` | tee /image-build-info.txt
 ENV X509_USER_PROXY /etc/grid-security/x509up
 
-CMD sh -c "python x509_updater.py --voms $VOMS"
+CMD sh -c "python3 x509_updater.py --voms $VOMS"
```
This `CMD` is ignored in the helm chart and is set in the x509 deployment yaml. This PR will depend on a change in that chart

BenGalewsky commented:
Looks good. Requires [ServiceX PR 248](https://github.com/ssl-hep/ServiceX/pull/248) to work
